app/src/main/java/io/appium/uiautomator2/handler/SetClipboard.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.handler;
+
+import org.json.JSONObject;
+
+import io.appium.uiautomator2.handler.request.SafeRequestHandler;
+import io.appium.uiautomator2.http.AppiumResponse;
+import io.appium.uiautomator2.http.IHttpRequest;
+import io.appium.uiautomator2.server.WDStatus;
+import io.appium.uiautomator2.utils.ClipboardHelper;
+import io.appium.uiautomator2.utils.ClipboardHelper.ClipDataType;
+import io.appium.uiautomator2.utils.Logger;
+
+public class SetClipboard extends SafeRequestHandler {
+
+    public SetClipboard(String mappedUri) {
+        super(mappedUri);
+    }
+
+    @Override
+    public AppiumResponse safeHandle(IHttpRequest request) {
+        Logger.info("Set Clipboard command");
+        final Object content;
+        ClipDataType contentType = ClipDataType.TEXT;
+        String label = null;
+        try {
+            JSONObject payload = getPayload(request);
+            content = payload.get("content");
+            if (payload.has("contentType")) {
+                contentType = ClipDataType.valueOf(payload
+                        .getString("contentType")
+                        .toUpperCase());
+            }
+            if (payload.has("label")) {
+                label = payload.getString("label");
+            }
+            switch (contentType) {
+                case TEXT: ClipboardHelper.setTextData(label, (String) content);
```

Comment:
since the user can set contentType to something else, should we throw if it's not supported?

Reply:
it will raise above in `ClipDataType.valueOf`

Reply:
ok